plugins/slack_app/test/samson_slack_app/samson_plugin_test.rb

```diff
@@ -0,0 +1,42 @@
+require_relative '../test_helper'
+
+SingleCov.covered! unless defined?(Rake) # rake preloads all plugins
+
+describe SamsonSlackApp do
+  let(:deploy) { deploys(:succeeded_test) }
+  let(:stage) { deploy.stage }
+
+  describe 'while configured' do
+    with_env SLACK_CLIENT_ID: 'abc', SLACK_CLIENT_SECRET: 'def' do
+      describe :before_deploy do
+        it "sends notification on before hook" do
+          SlackMessage.any_instance.expects(:deliver)
+          Samson::Hooks.fire(:before_deploy, deploy, nil)
+        end
+      end
+
+      describe :after_deploy do
+        it "sends notification on after hook" do
+          SlackMessage.any_instance.expects(:deliver)
+          Samson::Hooks.fire(:after_deploy, deploy, nil)
+        end
+      end
+    end
+  end
+
+  describe 'while not configured' do
```

Comment:
prefer 1 test group per thing so it's easier to later go in and ignore everything expect what you are working on

```Ruby
describe :before_deploy
  it x
  it y
end

describe :after_deploy
  it x
  it y
end
```